module/zfs/zio.c

```diff
@@ -3650,6 +3661,16 @@ zio_ddt_child_write_ready(zio_t *zio)
 	int p = DDT_PHYS_FOR_COPIES(ddt, zio->io_prop.zp_copies);
 	ddt_phys_variant_t v = DDT_PHYS_VARIANT(ddt, p);
 
+	if (ddt_phys_is_gang(dde->dde_phys, v)) {
+		for (int i = 0; i < BP_GET_NDVAS(zio->io_bp); i++) {
+			dva_t *d = &zio->io_bp->blk_dva[i];
+			metaslab_group_alloc_decrement(zio->io_spa,
+			    DVA_GET_VDEV(d), METASLAB_ASYNC_ALLOC,
+			    zio->io_allocator, B_FALSE, zio);
```

Comment:
The arguments list here is totally messed up, fitting neither the old nor the new.  I guess it might be the cause of the CI panic due to `mga_queue_depth` leak here.